Content.Server/Storage/EntitySystems/StorageSystem.Fill.cs

```diff
@@ -19,13 +17,29 @@ private void OnStorageFillMapInit(EntityUid uid, StorageFillComponent component,
         }
 
         var coordinates = Transform(uid).Coordinates;
-        var alreadySpawnedGroups = new HashSet<string>();
 
+        var orGroupedSpawns = new Dictionary<string, OrGroup>();
+
+        // collect groups together, create singular items that pass probability
         foreach (var entry in component.Contents)
         {
             // Handle "Or" groups
-            if (!string.IsNullOrEmpty(entry.GroupId) && alreadySpawnedGroups.Contains(entry.GroupId)) continue;
+            if (!string.IsNullOrEmpty(entry.GroupId))
+            {
+                if (!orGroupedSpawns.TryGetValue(entry.GroupId, out OrGroup? orGroup))
+                {
+                    OrGroup currentGroup = new(new List<EntitySpawnEntry>(), 0f);
+                    currentGroup.Entries.Add(entry);
+                    currentGroup.CumulativeProbability += entry.SpawnProbability;
+                    orGroupedSpawns.Add(entry.GroupId, currentGroup);
+                    continue;
```

Comment:
```suggestion
                    orGroup = new(new List<EntitySpawnEntry>(), 0f);
                    orGroupedSpawns.Add(entry.GroupId, orGroup);
```

Should work

Reply:
made the recommended changes and tested, looks like everything's still working as intended